src/python/WMCore/MicroService/MSOutput/RelValPolicy.py

```diff
@@ -102,22 +119,95 @@ def _validatePolicy(self, policyDesc, validDBSTiers, validDiskRSEs):
+
+        [{"releaseType": "pre", "lifetimeSecs": 120},
+         {"releaseType": "default", "lifetimeSecs": 360}]
+        :param lifeTDesc: list of dictionaries with the lifetime policy definition
+        :return: nothing, but it will raise an exception if any validation fails
+        """
+        if not isinstance(lifeTDesc, list):
+            msg = "The RelVal lifetime output data placement policy is not in the expected data type. "
+            msg += "Type expected: list, while the current data type is: {}. ".format(type(lifeTDesc))
+            msg += "This critical ERROR must be fixed."
+            raise RelValPolicyException(msg) from None
+
+        # policy must have a default/fallback destination for non pre-releases
+        setRelTypes = set()
+        expRelTypesKeys = {"pre", "default"}
+        for item in lifeTDesc:
+            if not isinstance(item.get('releaseType', None), str):
+                msg = "The 'releaseType' parameter must be a string, not {}.".format(type(item['releaseType']))
+                raise RelValPolicyException(msg) from None
+            if item['releaseType'] not in expRelTypesKeys:
+                msg = "The 'releaseType' parameter does not match the expected values. "
+                msg += "Value provided '{}' not in {}.".format(item['releaseType'], expRelTypesKeys)
+                raise RelValPolicyException(msg) from None
+            if not isinstance(item['lifetimeSecs'], int):
+                msg = "The 'lifetimeSecs' parameter must be integer, not {}".format(type(item['lifetimeSecs']))
+                raise RelValPolicyException(msg) from None
+            setRelTypes.add(item['releaseType'])
+
+        # last check, it must contain a policy for "pre" releases and "default"
+        if setRelTypes != expRelTypesKeys:
+            msg = "Policy must define rules for these 2 sample types: {}".format(expRelTypesKeys)
+            raise RelValPolicyException(msg) from None
+
+    def _convertTierPolicy(self, policyDesc):
+        """
+        Maps the RelVal tier data policy to a flat dictionary key'ed by datatiers
+        :param policyDesc: list of dictionaries with the tier policy definition
+        :return: a dictionary with a map of the RelVal tier policy
         """
         outputPolicy = dict()
         for item in policyDesc:
             outputPolicy.update({item['datatier']: item['destinations']})
         return outputPolicy
 
+    def _convertLifePolicy(self, policyDesc):
+        """
+        Maps the RelVal lifetime data policy to a flat dictionary key'ed
+        by the release cycle type (only supports pre or anything else).
+        :param policyDesc: list of dictionaries with the lifetime policy definition
+        :return: a dictionary with a map of the RelVal lifetime policy
+        """
+        outputPolicy = dict()
+        for item in policyDesc:
+            outputPolicy.update({item['releaseType']: item['lifetimeSecs']})
```

Comment:
Even though it may be not the case here, but are you sure that you should not check for existing key in a dict. In other words if your data input contains different values for the same releaseType only one (last) will be written to the dict.

Reply:
It's already done. This method is only executed after the input data schema is validated, with the method `_validateLifetimePolicy`.

Reply:
if so, you should probably update the docstring to reflect that.

Reply:
if so, you should probably update the docstring to reflect that.

Reply:
This method gets called right after the method performing the data sanitization/schema. IMO, there is no gain in mention it here again. Please have a look at the `__init__` method.